### ISSUE TITLE
"single-element of an array" model

### ISSUE THREAD

**Issue by xmh0511** [cplusplus/draft#4797]
[basic.compound] p3
> an object of type T that is not an array element is considered to belong to an array with one element of type T. 

Does it mean this rule also applies to any subobject that is not an array element? It does not make sense, I also think it's not the intent of the rule. The free room to construe this rule could result in https://github.com/cplusplus/draft/issues/3614.

Could we rephrase it to that? 
> a **complete** object of type T is considered to be the **initial element** of an array with one element of type T.

**Comment by JohelEGP** [comment]
If the intent isn't to make `std::views::single(`_`suboject`_`)` a valid range, where is the wording that would make it such?

**Comment by xmh0511** [comment]
Yes, you're right. That's the exact intent.
